### PR TITLE
fixes 'kfctl downloading the kubeflow to a local cache should be idempotent'

### LINKED ISSUE #2818: kfctl downloading the kubeflow repo to a local cache should be idempotent
I've seen the download to cache fail half-way through. It `kfctl init` is rerun we shouldn't return an error if the cache exists, rather just replace it.

### PR DESCRIPTION
**Pull request by kkasravi**
fixes #2818

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2819)

### CODE REVIEW

File: bootstrap/pkg/kfapp/coordinator/coordinator.go [hunk 1]
```diff
@@ -76,7 +76,10 @@ func downloadToCache(platform string, appDir string, version string, useBasicAut
 	cacheDir := path.Join(appDir, kftypes.DefaultCacheDir)
 	// idempotency
 	if _, err := os.Stat(cacheDir); !os.IsNotExist(err) {
-		os.RemoveAll(cacheDir)
+		rmErr := os.RemoveAll(cacheDir)
```
Comment (gabrielwen) [inline bootstrap/pkg/kfapp/coordinator/coordinator.go]: is this gonna be used by click to deploy app?  if it is, need to be careful of thread safety.

@kunmingg

Comment (jlewi) [reply bootstrap/pkg/kfapp/coordinator/coordinator.go]: What's the threading issue here? 

My expectation is each request in the deploy app uses a separate instance of KfApp and uses a different directory to copy down the configs to.

So even if we simultanteously handle two duplicate requests for the same app; both would be modifying a different directory.

Comment (gabrielwen) [reply bootstrap/pkg/kfapp/coordinator/coordinator.go]: yea, using different directory would be fine.  just wanna make sure it's working that way.

**Comment by jlewi** [comment]
/lgtm
/approve
/hold
@kunmingg Any concerns about use in deploy app?

File: bootstrap/pkg/kfapp/coordinator/coordinator.go (hunk 1 above)
Comment (kunmingg) [reply bootstrap/pkg/kfapp/coordinator/coordinator.go]: Shouldn't affect logic of deploy app. thanks

**Comment by jlewi** [comment]
Most recent failures are kubeflow/testing#342

**Comment by jlewi** [comment]
In the recent test it looked like delete failed.

/test all

**Pull request closed without merging**